admin-perspective/olm/install-forcenamespace/install-forcenamespace.md

```diff
@@ -0,0 +1,43 @@
+# Operator-defined Namespace
+
+Operator bundles can define what namespace they should be installed to, so that namespace is recommended to the user. The user can still change the installation namespace if desired.
+
+The installed namespace is always conveyed to the user for understanding of the underlying behavior.
+
+## No operator-defined namespace (existing behavior)
+
+![Operator namespace not defined all](img/1-1-notarget-all.png)
+- Namespace selection is read-only and fixed to openshift-operators.
+
+![Operator namespace not defined single](img/1-2-notarget-single.png)
+- All namespaces are available for installation.
+
+## Operator-defined namespace - already exists
+
+![Operator namespace defined all](img/2-1-targetExists-all.png)
+- Namespace selection defaults to operator-defined namespace with option to change to openshift-operators.
+
+![Operator namespace defined single](img/2-2-targetExists-single.png)
+- Namespace selection defaults to operator-defined namespace with option to change to any other namespace.
+
+## Operator-defined namespace - does not exist
+
+![Operator namespace not created all](img/3-1-targetNotExist-all.png)
+- Namespace selection defaults to operator-defined namespace with option to change to openshift-operators.
+- When the not yet created operator-defined namespace is selected, the user is alerted that the namespace will be created.
+
+![Operator namespace not created single](img/3-2-targetNotExist-single.png)
+- Namespace selection defaults to operator-defined namespace with option to change to any other namespace
+- When the not yet created operator-defined namespace is selected, the user is alerted that the namespace will be created.
```

Comment:
While implementing this scenario (and also 4-2) I've hit an limitation of our namespace dropdown `<NsDropdown>` component, where we can't pass to the component any additional options, in this case a namespace that operator recommends and does not exists. 
Since this is an only case I would rather take a moment and try to redesign.

Due to this limitation I've created additional solution that should cover all the use-cases:

AllNamespace install mode with recommended targetNamespace, where it does not exists:
<img width="862" alt="Screen Shot 2020-01-03 at 14 52 15" src="https://user-images.githubusercontent.com/1668218/71727284-0e3bc380-2e3a-11ea-921d-6a74579da9ef.png">


AllNamespace install mode with recommended targetNamespace, but picked `openshift-operators`, which exists:
<img width="861" alt="Screen Shot 2020-01-03 at 14 54 01" src="https://user-images.githubusercontent.com/1668218/71727285-0e3bc380-2e3a-11ea-9567-dfb11c75ee47.png">


OwnNamespace install mode with recommended targetNamespace, where it does not exists:
<img width="877" alt="Screen Shot 2020-01-03 at 15 07 16" src="https://user-images.githubusercontent.com/1668218/71727515-c79a9900-2e3a-11ea-8657-7fffa72d422f.png">


OwnNamespace install mode with recommended targetNamespace, but picked one of the existing namespaces:
<img width="913" alt="Screen Shot 2020-01-03 at 14 54 28" src="https://user-images.githubusercontent.com/1668218/71727288-0e3bc380-2e3a-11ea-80d8-3ecf27c3c86b.png">


In case the recommended targetNamespace wont be defined by the operator, the `Installed Namespace` section will only contain the `<NsDropdown>` component with all the available namespaces.
In case of:
- AllNamespace install mode - the `openshift-operators` namespace will be selected in dropdown component, and it will be disabled
- OwnNamespace install mode - the component will be enabled to user to pick any of the available namespaces.


Would really like your input @itsptk @bparees

Reply:
Have updated design to include separate radio in 'Installed Namespace' section to select targetNamespace to work around this dropdown issue.

Reply:
> While implementing this scenario (and also 4-2) I've hit an limitation of our namespace dropdown <NsDropdown> component, where we can't pass to the component any additional options, in this case a namespace that operator recommends and does not exists.

@jhadvig I think we can make this work if we need to.

Reply:
i think the "enable monitoring" checkbox should always be shown for any operator that requested monitoring in its CSV, regardless of what namespace it is being installed into (already exists or not, operator recommended or not, etc).

The only caveats i'd put on that are:
1) don't want them to enable it on `openshift-operators` since that's clearly community
2) enabling it on a namespace that isn't in the cluster-monitoring whitelist (i.e. openshift-*, kube-*, a few others) won't actually do anything, so it might be nice to not let people do that.

Reply:
> i think the "enable monitoring" checkbox should always be shown for any operator that requested monitoring in its CSV, regardless of what namespace it is being installed into (already exists or not, operator recommended or not, etc).

@bparees so form should basically create or update the namespace, right ?

Reply:
> enabling it on a namespace that isn't in the cluster-monitoring whitelist (i.e. openshift-, kube-, a few others)

do we have a complete list of the "others" ? if not, will it be sufficient to whitelist  openshift-, kube- namespaces ?

Reply:
@jhadvig yes the form should
1) create the namespace if needed
2) label the namespace if needed (operator requested monitoring and the checkbox is checked)

Reply:
> do we have a complete list of the "others" ? if not, will it be sufficient to whitelist openshift-, kube- namespaces ?

@LiliC @s-urbaniak, list of namespaces that cluster monitoring watches?  I think it's openshift-*, kube-*, and "default" but i'm not positive.

Reply:
Others are `kube-*`, `default`, and `logging` as per https://github.com/openshift/cluster-monitoring-operator/blob/3d719a9081a5db80e8da488af18d6a745c9b2927/jsonnet/main.jsonnet#L132.

Reply:
fwiw we are also having an epic to loosen that restriction and allow any namespace to be monitored via cluster-monitoring, iff it has the monitoring label set. That most likely won't happen though before 4.5.

Reply:
Ok so in summary, these would be the controls that appear when operator requested monitoring in its CSV:
![enable monitoring](https://user-images.githubusercontent.com/54860125/72106082-b8b15c00-32fc-11ea-9e31-ebdd6afa7a6f.png)

If monitoring was not requested in the operator's CSV no checkbox would appear.

Reply:
@itsptk other than the fact that we're going to probably need to strengthen the warning/note, looks right to me.

Reply:
Looking at the design again, I realize this reintroduces the issue of showing a possibly not-yet-created namespace in the availability namespace dropdown (Screen 3-2 & 5-1 of Jan 10th design.) Is the limitation of the dropdown component still an issue? @jhadvig